### ISSUE TITLE
issue about ofdm-transfer

### ISSUE THREAD

**Issue by liiso7** [righthalfplane/mystic#1]
Hello, I am using ofdm-transfer recently. The project is now read-only. I have encountered the same problem you left in the issue page. I would like to ask how is your current progress on this project and whether it can be successfully transmitted using bladerf? I'm using:
-bladerf （tx and rx)
-ubuntu 22.04

**Comment by righthalfplane** [comment]
There are several versions of the program. There is one example of transmitting and receiving on the same machine using a bladeRF - I think that that is "ofdm_loopback05.grc" on -

https://github.com/righthalfplane/rfspace

uBuntu 22.04 was causing problems, but I think they are mostly cleared up - In the past, I had to destroy all copies of the SoapSDR stuff and gnuradio and install from source to get things going. 

I shifted to uBuntu 23.10 and got things working there, but if you install CubicSDR it screws things up. When adding programs, remember to always start with -

sudo apt-get update
sudo apt-get upgrade

It does work on uBuntu with the bladeRF - Unfortunately, there is a huge list of things that can break it. Give it a try - If it does not work - post the output - perhaps, I can tell what is wrong.

**Comment by liiso7** [comment]
hello,i mean the project is :[https://github.com/glv2/ofdm-transfer](url)
Do you remember? I saw your comment on the issue page and I had the same problem as you and wanted to know how you finally solved it.

**Comment by righthalfplane** [comment]
Yes, Now I know which one you are talking about. I worked on that quite a while. He claimed that it worked, but I tried his exact setup and it did not work. It used a large number of frequencies, but the SDR's gain drops as the number of frequencies increases and the bandwidth decreases (I found this out latter). Another problem is the DC-OFFSET, I do not remember that there was any correction for that. There are many parameters here, It could be that his devices just happen to work ? His program uses several thousand frequencies, the example that worked (ofdm_loopback05.grc) only uses 12. With a hardwired line it worked, but as soon as it was transmitted it failed

**Comment by liiso7** [comment]
The problem page states that imprecise clocks can cause frequency offsets, If the offset between the center frequencies of the sender and the receiver is bigger than 1% of the OFDM signal  bandwidth, the receiver will probably not be able to decode the frames. So I used the same external clock source to clock both BladerFs, but I didn't receive any data at the receiver. Have you tried something similar before, or do you know what the problem is? Thanks for your answer!
Also I see you mentioned that you received the same five lines. What parameters are used, and what is the format of the file sent?

**Comment by righthalfplane** [comment]
Yes, they say imprecise frequency is one of most important things, but I had no way to test it. It is good to know that that is not the direct problem and I have not seem the  DC-OFFSET problems with the bladeRF devices.

Is his program actually transmiting, I seem to remember it required a few changes to get it going ?

His program is based on the liquid-dsp examples "flexframesync_example.c" and framesync64_example. These programs work fine encoding and decoding the data, but as some as I transmit and receive the data they fail. Here is where I posted the problem.

https://github.com/jgaeddert/liquid-dsp/issues/234

Perhaps, this discussion should be moved to my google group - somebody there may have some insight.

https://github.com/righthalfplane/SdrGlut
